src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -0,0 +1,935 @@
+     * is requested so it may end up heve while is not in the call anymore, so 404 error would indicate that there's
+     * no such conference anymore, but there has been one since the conferenceId was retrieved.
+     *
+     * @return ArrayNode containing RX/TX stats
+     * @throws Exception during http communication
+     */
+    public ArrayNode retrieveConferenceCallMediaStats(int conferenceId) throws Exception {
+        try {
+            return doGet(buildHttpUrl(String.format(MEDIASTATS, conferenceId)), ArrayNode.class);
+        } catch (CommandFailureException cfe) {
+            if (cfe.getStatusCode() == 404) {
+                if(logger.isDebugEnabled()) {
+                    logger.debug("Conference " + conferenceId + " is not available anymore. Skipping media stats retrieval.");
+                }
+                return JsonNodeFactory.instance.arrayNode();
+            } else {
+                logger.error("Unable to find conference by id " + conferenceId);
+                throw cfe;
+            }
+        }
+    }
+
+    /**
+     * Request device reboot
+     *
+     * @throws Exception during http communication
+     */
+    private boolean reboot() throws Exception {
+        ObjectNode rebootRequest = JsonNodeFactory.instance.objectNode();
+        rebootRequest.put("action", "reboot");
+        boolean success = doPost(REBOOT, rebootRequest).getStatusCode().is2xxSuccessful();
+        if (success) {
+            sessionId = null;
+            internalDestroy();
+        }
+        return success;
+    }
+
+    /**
+     * Retrieve device system information, including hardware info and lan status
+     *
+     * @param statistics map to put values to
+     * @throws Exception during http communication
+     */
+    public void getSystemInfo(Map<String, String> statistics) throws Exception {
+        JsonNode response = doGet(buildHttpUrl(SYSTEM), JsonNode.class);
+        if (response == null) {
+            return;
+        }
+        addStatisticsProperty(statistics, "System#Serial Number", response.get("serialNumber"));
+        addStatisticsProperty(statistics, "System#Software Version", response.get("softwareVersion"));
+        addStatisticsProperty(statistics, "System#System State", response.get("state"));
+        addStatisticsProperty(statistics, "System#System Name", response.get("systemName"));
+        addStatisticsProperty(statistics, "System#System Uptime", response.get("uptime"));
+        addStatisticsProperty(statistics, "System#System Build", response.get("build"));
+        addStatisticsProperty(statistics, "System#System Reboot Needed", response.get("rebootNeeded"));
+        addStatisticsProperty(statistics, "System#Device Model", response.get("model"));
+        addStatisticsProperty(statistics, "System#Device Hardware Version", response.get("hardwareVersion"));
+
+        if (response.get("lanStatus") == null) {

```

Comment:
"response.get("lanStatus")" can be extracted into local variable and reused below.